drivers/spi/spi_stm32.c

```diff
@@ -0,0 +1,584 @@
+/*
+ * Copyright (c) 2016 BayLibre, SAS
+ * Copyright (c) 2018 Bobby Noelte
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ * @brief STM32 SPI implementation
+ * @defgroup device_driver_spi_stm32 STM32 SPI inplementation
+ *
+ * A common driver for STM32 SPI. SoC specific adaptions are
+ * done by device tree and soc.h.
+ *
+ * @{
+ */
+
+/** Compile if at least one device is activated
```

Comment:
This comment confused me for a while.

It is communicating that we should only compile (something? the file?) if one device is activated.

But actually it is only controlling whether codegen should generate code in this file.

Reply:
Do you have a better comment phrase?

E.g. "Stop code generation if no device is activated"

Reply:
That will do.

Reply:
To be clear, 

self.generator_globals['generate_code'] = False

just causes the python codegen snippets to be omitted, not the entire file to be omitted, right?

Reply:
Ah, never mind, I was able to test and understand my own question.

I misunderstood the documentation then.

Reply:
Then the comment is correct, or, technically we still compile this file, but it doesn't have any contents.

Reply:
I did not expect codegen to be able to behave like this, because I was using this model, which is more restrictive than what we are doing here.

"
Codegen transforms files in a very simple way: it finds chunks of Python code
embedded in them, executes the Python code, and inserts its output back into
the original file. The file can contain whatever text you like around the
Python code. It will usually be source code.
"

Reply:
> Then the comment is correct, or, technically we still compile this file, but it doesn't have any contents.

Yes